cmd/bundle/bind.go

```diff
@@ -0,0 +1,85 @@
+package bundle
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/bundle/phases"
+	"github.com/databricks/cli/libs/cmdio"
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/jobs"
+	"github.com/databricks/databricks-sdk-go/service/pipelines"
+	"github.com/spf13/cobra"
+)
+
+func newBindCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "bind KEY RESOURCE_ID",
+		Short:   "Bind bundle-defined resources to existing resources",
+		PreRunE: ConfigureBundleWithVariables,
+		Args:    cobra.ExactArgs(2),
+	}
+
+	var autoApprove bool
+	cmd.Flags().BoolVar(&autoApprove, "auto-approve", false, "Automatically approve the binding")
+
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		b := bundle.Get(cmd.Context())
+		r := b.Config.Resources
+		resourceType, err := r.FindResourceByConfigKey(args[0])
+		if err != nil {
+			return err
+		}
+
+		w := b.WorkspaceClient()
+		err = checkResourceExists(cmd.Context(), w, resourceType, args[1])
+		if err != nil {
+			return fmt.Errorf("%s with an id '%s' is not found, err: %w", resourceType, args[1], err)
+		}
+
+		ctx := cmd.Context()
+		if !autoApprove {
+			answer, err := cmdio.AskYesOrNo(ctx, "Binding to existing resource means that the resource will be managed by the bundle which can lead to changes in the resource. Do you want to continue?")
```

Comment:
We should confirm that we would not recreate the resource on the TF side. Especially for pipelines this can be very dangerous where underlying tables get dropped if we do.

Reply:
@pietern I've added a step which shows terraform plan after import and asks user to confirm it, otherwise reverts the import